webapp/src/features/intersections/map/map-legend.tsx

```diff
@@ -129,70 +99,155 @@ export const MapLegend = () => {
-            marginLeft: 5,
-            marginRight: 3,
+            margin: '5px',
           }}
         >
-          <div style={{ fontSize: 12 }}>{key}: </div>
-          <img src={value} style={{ height: 40, width: 30, marginLeft: 2 }}></img>
-          <div>|</div>
+          <img src={value} style={{ height: 40, width: 30 }} />
+          <Typography fontSize="small" sx={{ ml: 1 }}>
+            {key}
+          </Typography>
         </div>
       </React.Fragment>
     )
   }
 
   return (
-    <Accordion disableGutters defaultExpanded={false}>
-      <AccordionSummary>
-        <Typography variant="h5">Legend</Typography>
-      </AccordionSummary>
-      <AccordionDetails>
-        <Paper className="legend" style={{ userSelect: 'none' }}>
-          <Accordion disableGutters defaultExpanded={true}>
-            <AccordionSummary>
-              <Typography variant="h5">Signal Heads</Typography>
-            </AccordionSummary>
-            <AccordionDetails>
-              <div style={{ display: 'flex', flexDirection: 'row', overflowX: 'auto' }}>{signalHeadIconsList}</div>
-            </AccordionDetails>
-          </Accordion>
-
-          <Divider sx={{ borderRadius: 1 }} />
-
-          <Accordion disableGutters defaultExpanded={true}>
-            <AccordionSummary>
-              <Typography variant="h5">Lane Lines</Typography>
-            </AccordionSummary>
-            <AccordionDetails>
-              <div style={{ display: 'flex', flexDirection: 'row', overflowX: 'auto' }}>{laneColorsList}</div>
-            </AccordionDetails>
-          </Accordion>
-
-          <Divider sx={{ borderRadius: 1 }} />
-
-          <Accordion disableGutters defaultExpanded={true}>
-            <AccordionSummary>
-              <Typography variant="h5">Lane Connections</Typography>
-            </AccordionSummary>
-            <AccordionDetails>
-              <div style={{ display: 'flex', flexDirection: 'row', overflowX: 'auto' }}>
-                {travelConnectionColorsList}
-              </div>
-            </AccordionDetails>
-          </Accordion>
-
-          <Accordion disableGutters defaultExpanded={true}>
-            <AccordionSummary>
-              <Typography variant="h5">BSM Colors</Typography>
-            </AccordionSummary>
-            <AccordionDetails>
-              <div style={{ display: 'flex', flexDirection: 'row', overflowX: 'auto' }}>{bsmColorsList}</div>
-            </AccordionDetails>
-          </Accordion>
-
-          <Divider sx={{ borderRadius: 1 }} />
-        </Paper>
-      </AccordionDetails>
-    </Accordion>
+    <>
+      <Fab
+        id="map-legend-button"
+        style={{
+          position: 'absolute',
+          zIndex: 10,
+          top: theme.spacing(3),
+          right: theme.spacing(10),
+          backgroundColor: theme.palette.background.paper,
+        }}
+        size="small"
+        onClick={() => {
+          setOpen(!open)
+        }}
+      >
+        <VpnKeyOutlined />
```

Comment:
I don't love this as a legend icon

Reply:
Maybe something more like this (from [wyoroadinfo](https://map.wyoroad.info/wtimap/index.html)): 
<img width="26" alt="image" src="https://github.com/user-attachments/assets/aebacd17-d62d-4570-9a82-c8aa453d050b" />

Reply:
Updated to use an icon similar to the one you suggested